usaspending_api/etl/elasticsearch_loader_helpers/delete_data.py

```diff
@@ -285,36 +313,45 @@ def delete_awards(
     if award_keys_len == 0:
         logger.info(
             format_log(
-                f"No related awards found for deletion. Zero transaction docs found from which to derive awards.",
+                "No related awards found for deletion. Zero transaction docs found from which to derive awards.",
                 action="Delete",
                 name=task_id,
             )
         )
-        return 0
-    logger.info(
-        format_log(f"Derived {award_keys_len} award keys from transactions in ES", action="Delete", name=task_id)
-    )
-
-    deleted_award_kvs = _check_awards_for_deletes(award_keys, spark)
-    deleted_award_kvs_len = len(deleted_award_kvs)
-    if deleted_award_kvs_len == 0:
-        # In this case it could be an award's transaction was deleted, but not THE LAST transaction of that award.
-        # i.e. the deleted transaction's "siblings" are still in the DB and therefore the parent award should remain
+    else:
         logger.info(
-            format_log(
-                f"No related awards found will be deleted. All derived awards are still in the DB.",
-                action="Delete",
-                name=task_id,
-            )
+            format_log(f"Derived {award_keys_len} award keys from transactions in ES", action="Delete", name=task_id)
         )
+        deleted_award_kvs = _check_awards_for_deletes(award_keys, spark)
+        deleted_award_kvs_len = len(deleted_award_kvs)
+        if deleted_award_kvs_len == 0:
```

Comment:
Might could initialize this to zero before the `if/else` block it's contained, then move it out of the conditional to prevent the nested conditions.